File: JAVA/Level_2_Arrays/WaveArray.java
// https://www.interviewbit.com/problems/wave-array/

import java.io.*;
import java.util.*;

class Solution {
  public ArrayList<Integer> wave(ArrayList<Integer> A) {
    int i, tmp, N=A.size();
    Collections.sort(A);
    if(N%2 == 1) {
      N--;
    }
    for(i=0; i<N; i+=2) {
      tmp = A.get(i);
      A.set(i, A.get(i+1));
      A.set(i+1, tmp);
    }
    return A;
  }
}

class WaveArray {
  public static void main(String args[]) {
    Solution s = new Solution();
    int i;
    ArrayList<Integer> X = new ArrayList<Integer>();
    ArrayList<Integer> Y = new ArrayList<Integer>();
    X.add(8);
    X.add(7);
    X.add(6);
    X.add(5);
    X.add(4);
    X.add(3);
    X.add(2);
    X.add(1);
    X.add(0);
    Y = s.wave(X);
    for(i=0; i<Y.size(); i++) {
      System.out.print(Y.get(i)+" ");
    }
  }
}

File: C++/Level_2_Arrays/WaveArray.cc
#include<iostream>
#include<vector>
using namespace std;

class Solution {
  public:
    vector<int> wave(vector<int> &);
};

vector<int> Solution::wave(vector<int> &A) {
  int i, N = A.size();
  if(N%2) {
    N--;
  }
  sort(A.begin(), A.end());
  for(i=0; i<N; i+=2) {
    swap(A[i], A[i+1]);
  }
  return A;
}

int main(void ) {
  Solution s;
  vector<int> v, _v;
  size_t i;

  v.push_back(7);
  v.push_back(6);
  v.push_back(5);
  v.push_back(4);
  v.push_back(3);
  v.push_back(2);
  v.push_back(1);
  v.push_back(0);

  _v = s.wave(v);

  for(i=0; i<v.size(); i++){
    cout<<_v[i]<<' ';
  }

  return 0;
}
